From a2a36cbce801e49c892b458fab9f664581000b15 Mon Sep 17 00:00:00 2001
From: Martin Moya <moyamartin1@gmail.com>
Date: Thu, 19 Jun 2025 18:40:17 -0300
Subject: [PATCH] Set custom env variables to rpi.env

---
 board/raspberrypi/rpi/rpi.env | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
index 30228285edd..14a17c888e7 100644
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -75,3 +75,16 @@ fdt_addr_r=0x02600000
 ramdisk_addr_r=0x02700000
 
 boot_targets=mmc usb pxe dhcp
+rpipart=2
+bootlimit=5
+altbootcmd=" \
+    echo Rollback to previous RootFs; \
+    if test ${rpipart} = 2; \
+        then setenv rpipart 3; \
+    else \
+        setenv rpipart 2; \
+    fi; \
+    setenv bootcount 0; \
+    saveenv; \
+    bootcmd \
+"
---
2.43.0
